server/src/main/java/org/spine3/server/projection/ProjectionTransaction.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.spine3.server.projection;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.Message;
+import org.spine3.base.EventContext;
+import org.spine3.base.Version;
+import org.spine3.server.entity.Transaction;
+import org.spine3.server.entity.TransactionListener;
+import org.spine3.validate.ValidatingBuilder;
+
+import java.lang.reflect.InvocationTargetException;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A transaction, within which {@linkplain Projection projection instances} are modified.
+ *
+ * @param <I> the type of projection IDs
+ * @param <M> the type of projection state
+ * @param <B> the type of a {@code ValidatingBuilder} for the projection state
+ * @author Alex Tymchenko
+ */
+class ProjectionTransaction<I,
+                            M extends Message,
+                            B extends ValidatingBuilder<M, ? extends Message.Builder>>
+        extends Transaction<I, Projection<I, M, B>, M, B> {
+
+    @VisibleForTesting
+    ProjectionTransaction(Projection<I, M, B> entity) {
+        super(entity);
+    }
+
+    @VisibleForTesting
+    ProjectionTransaction(Projection<I, M, B> entity,
+                          TransactionListener<I, Projection<I, M, B>, M, B> listener) {
+        super(entity, listener);
+    }
+
+    @VisibleForTesting
+    ProjectionTransaction(Projection<I, M, B> entity, M state, Version version) {
+        super(entity, state, version);
+    }
+
+    @Override
+    protected void invokeApplier(Projection entity,
```

Comment:
One option I can see is `invokeHandler`, which would have the similar flaw.

The idea of this method is _just_ call the event applier/handler method, so it cannot be called `apply`, because nothing is actually applied in the descendants. 

I will think of another name.